File: frontend/ios/CapsuleAR/ARWorldMapViewManager.m
// ARWorldMapViewManager.m
// ObjC bridge declaration for ARWorldMapViewManager.swift.
// React Native's bridge is Objective-C, so every Swift RCTViewManager needs
// this one-liner to be visible to the bridge. No methods to expose here —
// the view manager just creates the view.

#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE(ARWorldMapViewManager, RCTViewManager)
@end

File: frontend/ios/CapsuleAR/ARWorldMapViewManager.swift
import React

@objc(ARWorldMapViewManager)
class ARWorldMapViewManager: RCTViewManager {

    override static func requiresMainQueueSetup() -> Bool {
        return true
    }

    override func view() -> UIView! {
        let arView = ARWorldMapView()

        // Link the view to the module so events flow through
        if let module = self.bridge.module(for: ARWorldMapModule.self) as? ARWorldMapModule {
            module.setARView(arView)
        }

        return arView
    }
}
